### PR TITLE
Add EBS volumes

### PR DESCRIPTION
**Pull request by maratoid**
replace regexp with ERB, cleanup node type assignment. use EBS drives, as ephemeral on m3 is too small

### CODE REVIEW

File: kubernetes/clusters/aws/Vagrantfile [hunk 1]
```diff
@@ -121,6 +121,34 @@ def get_elb_dns_info(aws_settings)
...
 
+def get_ami_type(node_info, aws_settings)
+  aws_instance_type = aws_settings['instanceType']
+  case node_info[:type]
+  when 'master'
+    aws_instance_type = aws_settings['instanceTypeMaster'] unless aws_settings['instanceTypeMaster'].nil?
+  when 'etcd'
+    aws_instance_type = aws_settings['instanceTypeEtcd'] unless aws_settings['instanceTypeEtcd'].nil?
+  when 'node'
+    aws_instance_type = aws_settings['instandeTypeNode'] unless aws_settings['instandeTypeNode'].nil?
```
Comment (spiffxp) [inline kubernetes/clusters/aws/Vagrantfile]: typos, should be instanceTypeNode

Comment (spiffxp) [inline kubernetes/clusters/aws/Vagrantfile]: `get_ami_type` is misleading, this should be `get_instance_type` I think?

This is also appears to be missing the special case for `instanceTypeNode01`

File: kubernetes/clusters/aws/Vagrantfile [hunk 2]
```diff
@@ -121,6 +121,34 @@ def get_elb_dns_info(aws_settings)
...
+
+def get_ebs_size(node_info, aws_settings)
+  ebs_size = aws_settings['instanceEBSSize'] || 300
+  case node_info[:type]
+  when 'master'
+    ebs_size = aws_settings['masterEBSSize'] unless aws_settings['masterEBSSize'].nil?
+  when 'etcd'
+    ebs_size = aws_settings['etcdEBSSize'] unless aws_settings['etcdEBSSize'].nil?
+  when 'node'
+    ebs_size = aws_settings['nodeEBSSize'] unless aws_settings['nodeEBSSize'].nil?
```
Comment (spiffxp) [inline kubernetes/clusters/aws/Vagrantfile]: Would also like to see a node01EBSSize

File: kubernetes/Vagrantfile [hunk 3]
```diff
@@ -176,44 +178,34 @@ def build_coreos_userdata(host_number)
...
-  end
+  render(
+    node_info[:user_data], 
+    tmp_user_data,
+    { 
+      :etcd_cluster_ip => etcd_cluster_ip, 
+      :master_cluster_ip => master_cluster_ip,
+      :kubernetes_release => 'v' + kubernetes_release,
+      :api_version => settings['cluster']['apiVersion'],
+      :kubernetes_verbocity => '2',
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: typo, this should be `kubernetes_verbosity`

File: kubernetes/Vagrantfile [hunk 4]
```diff
@@ -176,44 +178,34 @@ def build_coreos_userdata(host_number)
...
+      :reboot_strategy => settings['coreos']['rebootStrategy'] || 'off',
+      :cloud_provider => 'vagrant',
+      :dns_domain => 'kubernetes.local',
+      :hostname => node_info[:hostname],
+      :service_public_ip => proxy_cluster_ip,
+      :kraken_services_branch => cluster_services['branch'],
+      :kraken_services_repo => cluster_services['repo'],
+      # TODO: parse out from cluster.services
+      :kraken_services_grafana_enabled => 'true',
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: would prefer to see this going back to `kraken_services_influxdb_grafana_enabled`... eventually I will split influxdb into its own service, for now I'd rather not be misleading that grafana also launches influxdb

**Comment by spiffxp** [comment]
Some things I'd like to see addressed before merging this.  I'd like to hack in a way to configure whether it's EBS or ephemeral if I find EBS performance varies too wildly, going to need to test this for a bit once this is merged.

**Comment by maratoid** [comment]
I'll have something with all of the above + ebs vs ephemeral option

**Comment by l337ch** [comment]
With docker being on ext4 - is the docker option '--storage overlay' need to enable overlayfs?

**Comment by spiffxp** [comment]
Anecdotally I saw the 'overlay' fs show up inside of influxdb container, will double check

**Pull request merged**